cmd/kube-proxy/app/server.go

```diff
@@ -474,32 +474,33 @@ Service cluster IPs and ports are currently found through Docker-links-compatibl
 environment variables specifying ports opened by the service proxy. There is an optional
 addon that provides cluster DNS for these cluster IPs. The user must create a service
 with the apiserver API to configure the proxy.`,
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
 			verflag.PrintAndExitIfRequested()
 			cliflag.PrintFlags(cmd.Flags())
 
 			if err := initForOS(opts.WindowsService); err != nil {
 				klog.ErrorS(err, "Failed OS init")
```

Comment:
Ideally errors should be handled once - not logger AND returned (the called will log it too!). Would be better to augment the error with the string (if it's necessary) instead of logging. Then the caller will log it once.

Reply:
Thanks, fixed it.